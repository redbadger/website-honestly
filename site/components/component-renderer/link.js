// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
