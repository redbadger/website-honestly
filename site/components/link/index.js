// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
