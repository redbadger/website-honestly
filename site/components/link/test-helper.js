// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
