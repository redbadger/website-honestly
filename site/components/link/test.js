// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
