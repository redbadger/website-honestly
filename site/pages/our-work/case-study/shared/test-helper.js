// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
