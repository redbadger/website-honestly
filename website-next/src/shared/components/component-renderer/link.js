import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
