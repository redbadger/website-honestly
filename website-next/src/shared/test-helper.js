import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
