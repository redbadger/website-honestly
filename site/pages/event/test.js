import React from 'react';
import { render, mount } from 'enzyme';

import { Context } from '../../components/link/test-helper';

import Event from './index';
import EventbriteEmbed from './eventbrite-embed';

const event = {
  id: 'W00fw00F4mD0g3',
  startDateTime: {
    date: '31',
    month: '01',
    year: '2017',
    monthSym: 'January',
  },
  endDateTime: {
    date: '01',
    month: '02',
    year: '2017',
    monthSym: 'February',
  },
  title: 'I am the best event ever',
  strapline: 'Come along to the best event ever',
  body: [{ text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit' }],
  internalLinks: [
    {
      url: 'https://www.red-badger.com/some-event',
      title: 'some event',
    },
  ],
  externalLinks: [
    {
      url: 'https://www.external-site.com/some-event',
      title: 'some event',
    },
  ],
  ticketLink: 'https://www.eventbrite.com/some-event-123456789',
  eventbriteId: '123456789',
  featureImageFilename: 'i-am-an-image.jpg',
  location: {
    address: 'Badgerham Palace',
  },
};

describe('site/pages/event', () => {
  it('renders correctly', () => {
    expect(
      render(
        <Context>
          <Event event={event} />
        </Context>,
      ),
    ).toMatchSnapshot();
  });

  it('Renders eventbrite modal button for eventbrite events', () => {
    const eventPage = mount(
      <Context>
        <Event event={event} />
      </Context>,
    );

    expect(
      eventPage.contains(
        <EventbriteEmbed eventbriteId={event.eventbriteId} url={event.ticketLink} />,
      ),
    ).toBe(true);
  });

  it('Does not render eventbrite modal button for non-eventbrite events', () => {
    const newEvent = event;
    newEvent.eventbriteId = null;
    const eventPage = mount(
      <Context>
        <Event event={newEvent} />
      </Context>,
    );
    expect(
      eventPage.contains(
        <EventbriteEmbed eventbriteId={newEvent.eventbriteId} url={newEvent.ticketLink} />,
      ),
    ).toBe(false);
  });
});
