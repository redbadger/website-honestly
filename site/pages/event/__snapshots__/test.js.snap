// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/event renders correctly 1`] = `
<div
  class="background"
  itemscope=""
  itemtype="http://schema.org/Event"
>
  <span
    itemprop="location"
    itemscope=""
    itemtype="http://schema.org/Place"
  >
    <span
      content="Badgerham Palace"
      itemprop="address"
    />
  </span>
  <section
    class="section"
  >
    <div
      class="wrapper"
    >
      <div
        class="container"
      >
        <div
          class="mainContainer"
        >
          <hr
            class="hr grey mobileHorizontalLine"
          />
          <div
            class="dateBubble"
            itemprop="startDate"
          >
            31 January 2017 - 01 February 2017
          </div>
          <h1
            class="eventTitle"
            itemprop="name"
          >
            I am the best event ever
          </h1>
          <div
            class="twoColumn"
          >
            <div
              class="event"
            >
              <div
                class="eventDescription"
              >
                Come along to the best event ever
              </div>
              <div
                class="eventBody"
              >
                <p
                  class="paragraph"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </p>
              </div>
              <div
                class="eventbrite-modal"
              >
                <div>
                  <noscript>
                    &lt;a href="https://www.eventbrite.com/some-event-123456789" rel="noopener noreferrer" target="_blank"&gt;Get tickets on Eventbrite&lt;/a&gt;
                  </noscript>
                  <button
                    class="getTickets__button"
                    id="eventbrite-modal-123456789"
                    type="button"
                  >
                    Get tickets
                  </button>
                </div>
              </div>
              <div>
                <div
                  class="eventLinks"
                >
                  <div
                    class="eventLinkList cf"
                  >
                    <a
                      class="fullDetailsLink"
                      href="https://www.external-site.com/some-event"
                      rel="noopener"
                      target="_blank"
                    >
                      <span>
                        some event
                      </span>
                    </a>
                  </div>
                  <div
                    class="eventLinkList cf"
                  >
                    <a
                      class="fullDetailsLink"
                      href="https://www.red-badger.com/some-event"
                    >
                      <span>
                        some event
                      </span>
                    </a>
                  </div>
                </div>
              </div>
              <hr
                class="hr grey horizontal-line"
              />
              <div
                class="moreEvents"
              >
                <a
                  class=""
                  href="/events"
                >
                  <span>
                    More events
                  </span>
                </a>
              </div>
            </div>
            <div>
              <img
                alt="I am the best event ever"
                class="eventImg"
                role="presentation"
                src="i-am-an-image.jpg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
